static/app/views/settings/project/projectOwnership/addCodeOwnerModal.tsx

```diff
@@ -24,215 +24,209 @@ import type {
-    const {organization, project} = this.props;
-    const endpoints: ReturnType<DeprecatedAsyncComponent['getEndpoints']> = [
-      [
-        'codeMappings',
-        `/organizations/${organization.slug}/code-mappings/`,
-        {query: {project: project.id}},
-      ],
-      [
-        'integrations',
-        `/organizations/${organization.slug}/integrations/`,
-        {query: {features: ['codeowners']}},
-      ],
-    ];
-    return endpoints;
-  }
+  const {
+    data: integrations,
+    isPending: isIntegrationsPending,
+    isError: isIntegrationsError,
+  } = useApiQuery<Integration[]>(
+    [
+      `/organizations/${organization.slug}/integrations/`,
+      {query: {features: ['codeowners']}},
+    ],
+    {staleTime: Infinity}
+  );
 
-  fetchFile = async (codeMappingId: string) => {
-    const {organization} = this.props;
-    this.setState({
-      codeMappingId,
-      codeownersFile: null,
-      error: false,
-      errorJSON: null,
-      isLoading: true,
-    });
-    try {
-      const data: CodeownersFile = await this.api.requestPromise(
-        `/organizations/${organization.slug}/code-mappings/${codeMappingId}/codeowners/`,
-        {
-          method: 'GET',
-        }
-      );
-      this.setState({codeownersFile: data, isLoading: false});
-    } catch (_err) {
-      this.setState({isLoading: false});
-    }
-  };
+  const [codeMappingId, setCodeMappingId] = useState<string | null>(null);
 
-  addFile = async () => {
-    const {organization, project, onSave, closeModal} = this.props;
-    const {codeownersFile, codeMappingId, codeMappings} = this.state;
+  const {
+    data: codeownersFile,
+    isPending: isCodeownersFilePending,
+    isError: isCodeownersFileError,
+  } = useApiQuery<CodeownersFile>(
+    [`/organizations/${organization.slug}/code-mappings/${codeMappingId}/codeowners/`],
+    {staleTime: Infinity, enabled: Boolean(codeMappingId)}
+  );
 
-    if (codeownersFile) {
-      const postData: {
-        codeMappingId: string | null;
-        raw: string;
-      } = {
-        codeMappingId,
-        raw: codeownersFile.raw,
-      };
+  const api = useApi({
+    persistInFlight: false,
+  });
+  const mutation = useMutation<
+    TCodeownersData,
+    TCodeownersError,
+    TCodeownersVariables,
+    TCodeownersContext
+  >({
+    mutationFn: ([payload]: TCodeownersVariables) => {
+      return fetchMutation(api)([
+        'POST',
+        `/projects/${organization.slug}/${project.slug}/codeowners/`,
+        {},
+        payload,
+      ]);
+    },
+    gcTime: 0,
+  });
 
-      try {
-        const data = await this.api.requestPromise(
-          `/projects/${organization.slug}/${project.slug}/codeowners/`,
-          {
-            method: 'POST',
-            data: postData,
+  const addFile = useCallback(() => {
+    if (codeownersFile) {
+      mutation.mutate([{codeMappingId, raw: codeownersFile.raw}], {
+        onSuccess: d => {
```

Comment:
Usually we put these `onSuccess` and `onError` handlers in the original `useMutation`, but this will have the same effect